core/src/packet_deduper.rs

```diff
@@ -0,0 +1,58 @@
+use {
+    crate::{banking_stage::BankingStageStats, packet_hasher::PacketHasher},
+    lru::LruCache,
+    solana_measure::measure::Measure,
+    solana_perf::packet::PacketBatch,
+    std::{
+        ops::DerefMut,
+        sync::{atomic::Ordering, Arc, Mutex},
+    },
+};
+
+const DEFAULT_LRU_SIZE: usize = 200_000;
+
+#[derive(Clone)]
+pub struct PacketDeduper(Arc<Mutex<(LruCache<u64, ()>, PacketHasher)>>);
+
+impl Default for PacketDeduper {
+    fn default() -> Self {
+        Self(Arc::new(Mutex::new((
+            LruCache::new(DEFAULT_LRU_SIZE),
+            PacketHasher::default(),
+        ))))
+    }
+}
+
+impl PacketDeduper {
+    pub fn dedupe_packets(
+        &self,
+        packet_batch: &PacketBatch,
+        packet_indexes: &mut Vec<usize>,
+        banking_stage_stats: &BankingStageStats,
```

Comment:
nit pick, functionally `PacketDeduper` doesn't need to couple with `BankingStageStats`. It could just return elapsed and count via two mutable params. 
Or Deduper can reports these two stats itself, help to reduce BankingStageStats size.

Reply:
I'd like to follow up on the stats nit picks in another PR because I'm already working on organizing banking stage stats in a better way, cool with you?

Reply:
Totally! Also, I have PR #21708 that tries to organize some of leader metrics from QoS perspective. Just want to bring it up so we won't overlap too much.